modules/category/category.service.js

```diff
@@ -18,6 +18,17 @@ class CategoryService {
   }
 
   async addCategory(data) {
+    if (await this.checkCategoryEXist(data)) {
+      throw new Error(CATEGORY_ALREADY_EXIST);
+    }
+    return new Category(data).save();
+  }
+
+  async deleteCategory(id) {
+    return await Category.findByIdAndDelete(id);
+  }
+
+  async checkCategoryExist(data) {
     const category = await Category.find({
```

Comment:
Use the method `count` here to reduce the amount of data that is transferred between app and db